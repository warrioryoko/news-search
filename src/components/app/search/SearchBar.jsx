import React from 'react';
import PropTypes from 'prop-types';

const SearchBar = ({ search, onSubmit, onChange }) => (
  <>
    <label htmlFor="search">Search here!</label>
    <input
      id="search"
      type="search"
      name="search"
      value={search}
      onChange={onChange}
    />
    <button onClick={onSubmit} data-testid="dasbutton">Push me!</button>
  </>
);

SearchBar.propTypes = {
  search: PropTypes.string.isRequired,
  onSubmit: PropTypes.func.isRequired,
  onChange: PropTypes.func.isRequired
};

export default SearchBar;
