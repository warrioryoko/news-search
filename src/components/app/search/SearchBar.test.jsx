import React from 'react';
import { render, cleanup } from '@testing-library/react';
import SearchBar from './SearchBar';

describe('Does the SearchBar component render?', () => {
  afterEach(() => cleanup());
  const dummyFunction = function() {};

  it('should render a SearchBar component', () => {
    const { asFragment } = render(<SearchBar
      search="Anythingdoesntmattererhmahgawd"
      onSubmit={dummyFunction}
      onChange={dummyFunction}
    />);
    expect(asFragment()).toMatchSnapshot();
  });
});
