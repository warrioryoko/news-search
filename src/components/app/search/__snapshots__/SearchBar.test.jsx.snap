// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Does the SearchBar component render? should render a SearchBar component 1`] = `
<DocumentFragment>
  <label
    for="search"
  >
    Search here!
  </label>
  <input
    id="search"
    name="search"
    type="search"
    value="Anythingdoesntmattererhmahgawd"
  />
  <button
    data-testid="dasbutton"
  >
    Push me!
  </button>
</DocumentFragment>
`;
